woocommerce/woocommerce
# fix: wrong condition for shipping recommendations tour step

### All Submissions:






### Changes proposed in this Pull Request:

One of the conditions for **test case 3** was invalid and has been updated to rely on a CSS selector for the element that it references. 

Note that the step count may show inconsistency at the start if loading takes awhile, but it updates after a step progression. A fix for that would involve watching the page for changes and that's too complex for a PR that violates the code freeze period.

Built WooCommerce zip here: https://github.com/woocommerce/woocommerce/suites/7573943138/artifacts/313201220

Closes #34039.

### How to test the changes in this Pull Request:

**Case 1: The store sells physical products and is located in the US, but JP and WCS are not installed.**

1. Start with a fresh install of WooCommerce
2. Start OBW and choose `United States` as store country
3. Choose "Physical products" 
4. Complete the OBW without installing anything from the `Business Details` tab.
5. From the WooCommerce homepage click on "Review Shipping" task
6. "United States (US)" zone should be created with `Free shipping` method
7. **There should be 3 steps in the tour, covering shipping zones, shipping methods and recommended shipping options**
8. The tour should not appear again upon revisit or refresh of the page, if you complete the tour

**Case 2: The store sells physical products, has JP and WCS installed and connected, and is located in the US.**

1. Start with a fresh install of WooCommerce (alternatively just go and delete the option for `woocommerce_admin_reviewed_default_shipping_zones`)
2. Start OBW and choose `United States` as store country
3. Choose "Physical products" 
4. Install Jetpack and WooCommerce Shipping from the `Business Details` tab.
5. Complete the OBW 
6. Connect and approve Jetpack when prompted.
7. From the WooCommerce homepage click on "Review Shipping" task
9. "United States (US)" zone should be created with `Free shipping` method
8. **There should be 3 steps in the tour, with the third one highlighting the WooCommerce Shipping section**
9. The tour should not appear again upon revisit or refresh of the page, if you complete the tour

**Case 3: The store sells physical products, and is not located in US/Canada/Australia/UK (irrelevant if JP is installed or not).**

1. Start with a fresh install of WooCommerce
2. Start OBW and choose a country that is not one of the US, Canada, Australia, and UK
3. Choose "Physical products" 
4. Complete the OBW without installing anything from the `Business Details` tab.
5. From the WooCommerce homepage click on "Review Shipping" task
6. "Country-you-chose" zone should be created with `Free shipping` method
7. **There should be 2 steps in the tour, covering shipping zones and shipping methods**
8. The tour should not appear again upon revisit or refresh of the page, if you complete the tour

**Case 4: The store sells digital products**

1. Start with a fresh install of WooCommerce
2. Start OBW and choose any country
3. Choose "Downloads" 
4. Complete the OBW without installing anything from the `Business Details` tab.
5. Navigate to WooCommerce -> Settings -> Shipping
6. You should not see any automatically created shipping settings and there should not be a tour

### Other information:




### FOR PR REVIEWER ONLY:
